server/bootserver.go

```diff
@@ -83,41 +87,27 @@ func handleIPXE(w http.ResponseWriter, r *http.Request, k8sClient client.Client,
 	log.Info("Processing IPXE request", "method", r.Method, "path", r.URL.Path, "clientIP", r.RemoteAddr)
 	ctx := r.Context()
 
-	clientIP, _, err := net.SplitHostPort(r.RemoteAddr)
-	if err != nil {
-		log.Error(err, "Failed to parse client IP address", "clientIP", r.RemoteAddr)
-		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
-		return
-	}
-
-	clientIPs := []string{clientIP}
-	if xff := r.Header.Get("X-Forwarded-For"); xff != "" {
-		for _, ip := range strings.Split(xff, ",") {
-			trimmedIP := strings.TrimSpace(ip)
-			if trimmedIP != "" {
-				clientIPs = append(clientIPs, trimmedIP)
-			}
+	if strings.HasPrefix(r.URL.Path, "/ipxe/") {
+		uuid := strings.TrimPrefix(r.URL.Path, "/ipxe/")
+		if uuid == "" {
+			http.Error(w, "Bad Request: UUID is required", http.StatusBadRequest)
+			return
 		}
-	}
 
-	var ipxeConfigs bootv1alpha1.IPXEBootConfigList
-	for _, ip := range clientIPs {
-		if err := k8sClient.List(ctx, &ipxeConfigs, client.MatchingFields{bootv1alpha1.SystemIPIndexKey: ip}); err != nil {
-			log.Info("Failed to list IPXEBootConfig for IP", "IP", ip, "error", err)
-			continue
+		ipxeBootConfigList := &bootv1alpha1.IPXEBootConfigList{}
+		err := k8sClient.List(ctx, ipxeBootConfigList, client.MatchingFields{bootv1alpha1.SystemUUIDIndexKey: uuid})
+		if client.IgnoreNotFound(err) != nil {
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
 		}
 
-		if len(ipxeConfigs.Items) > 0 {
-			log.Info("Found IPXEBootConfig", "IP", ip)
-			break
+		if len(ipxeBootConfigList.Items) == 0 {
+			log.Info("No specified UUID is found, delivering default script")
```

Comment:
I recommend replying with `404 Not Found` if IPXEBootConfig is not found for a related UUID instead of default script.

Reply:
I agree, added